# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/AsyncPanZoomController.cpp
# Commit: 496c75dc1301
# Full Hash: 496c75dc13011159293cc31ea7555d8835545c06
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2021-03-18 04:15:43
# Description:
#   Bug 1698841. Ignore non-touch input blocks in AsyncPanZoomController::OnDoubleTap. r=botond
#   
#   We can either get here via touch events interpreted as a double tap, or a double tap event sent from the os. The former will have a touch block, the latter will have no input block associated with it.
#   
#   The crash shows it's possible to get here with another type of current input block.
# ==============================================================================

diff -r 5a43d9f3ed4c -r 496c75dc1301 gfx/layers/apz/src/AsyncPanZoomController.cpp
--- a/gfx/layers/apz/src/AsyncPanZoomController.cpp	Wed Mar 17 22:50:46 2021 +0200
+++ b/gfx/layers/apz/src/AsyncPanZoomController.cpp	Wed Mar 17 20:47:57 2021 +0000
@@ -2895,9 +2895,8 @@
   APZC_LOG("%p got a double-tap in state %d\n", this, mState);
   RefPtr<GeckoContentController> controller = GetGeckoContentController();
   if (controller) {
-    MOZ_ASSERT(!GetCurrentInputBlock() || GetCurrentTouchBlock());
     if (mZoomConstraints.mAllowDoubleTapZoom &&
-        (!GetCurrentInputBlock() ||
+        (!GetCurrentTouchBlock() ||
          GetCurrentTouchBlock()->TouchActionAllowsDoubleTapZoom())) {
       if (Maybe<LayoutDevicePoint> geckoScreenPoint =
               ConvertToGecko(aEvent.mPoint)) {
